### DISCUSSION TITLE
Formas de padronizar códigos de um projeto em equipe.

### DISCUSSION THREAD

**Discussion by romulosardinha** [frontendbr/forum#1741] [Dicas]
Esta árdua tarefa de fazer várias pessoas seguirem alguns padrões as vezes é difícil.
Como vocês trabalham com isso? Deixo aqui minha atual visão sobre e gostaria de conhecer outras formas.

[https://medium.com/@romulosardinha/padronizando-projeto-em-equipe-remota-usando-react-vscode-e-blueprint-91691c194cab](https://medium.com/@romulosardinha/padronizando-projeto-em-equipe-remota-usando-react-vscode-e-blueprint-91691c194cab)

**Comment by ninetails** [comment]
Automatização realmente é uma ótima solução, o que acaba em tooling.

Eu realmente não cheguei a nenhum momento em ter que usar **code generator** pois acabo priorizando outras coisas como **styleguide**. Creio que a receita de sempre:

Configurar Husky para:
- não **commitar** código fora do padrão
- não **subir** código com testes falhando

Gerar código pessoalmente acho que o projeto deveria estar mais "maduro" para o tal... Mas desconhecia esse blueprint, eu já tentei brincar um pouco utilizando o [Plop](https://plopjs.com/) para ter algo mais desacoplado do VSCode direto no CLI.

Mas algo restrito ao VSCode, geralmente adiciono extensões como [Toggle excluded files](https://marketplace.visualstudio.com/items?itemName=eamodio.toggle-excluded-files) para "limpar" a árvore de arquivos escondendo o que for de configuração e estou vendo de deixar snippets por projeto. E deixar configurações como do `eslint` para formatar ao salvar, junto com a recomendação da extensão [mais famosinha](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint).
